module/data/item/templates/ability.mjs

```diff
@@ -210,6 +230,32 @@ export default class AbilityTemplate extends ActionTemplate.mixin(
     if ( !createData?.system?.level ) itemData.system.level = 0;
     const learnedItem = ( await actor.createEmbeddedDocuments( "Item", [ itemData ] ) )?.[0];
 
+    let category = null;
+    if ( item.type === "talent" ) {
```

Comment:
A line like this is good indication that inheritance should be used. So this entire block should be part of `TalentData`'s `learn` method instead of here.

Reply:
Sure, let's do that